library/src/blas1/rocblas_amax_amin.h

```diff
@@ -176,34 +142,25 @@ static rocblas_status rocblas_iamaxmin(
 
 extern "C" {
 
-rocblas_status JOIN(rocblas_isa, MAX_MIN)(
-    rocblas_handle handle, rocblas_int n, const float* x, rocblas_int incx, rocblas_int* result)
-{
-    return rocblas_iamaxmin<float>(handle, n, x, incx, result);
-}
+#ifdef IMPL
+#error IMPL IS ALREADY DEFINED
+#endif
 
-rocblas_status JOIN(rocblas_ida, MAX_MIN)(
-    rocblas_handle handle, rocblas_int n, const double* x, rocblas_int incx, rocblas_int* result)
-{
-    return rocblas_iamaxmin<double>(handle, n, x, incx, result);
-}
+#define IMPL(name_, typei_, typew_)                                   \
+    rocblas_status name_(rocblas_handle handle,                       \
+                         rocblas_int    n,                            \
+                         const typei_*  x,                            \
+                         rocblas_int    incx,                         \
+                         rocblas_int*   results)                      \
+    {                                                                 \
+        return rocblas_iamaxmin<typew_>(handle, n, x, incx, results); \
+    }
```

Comment:
Opinion: I prefer to see more code rather than less code if it is more straight forward, i.e. duplicated code that I can read I think is more maintainable than macros I can not.